### PR TITLE
Stop doing inventory refresh as part of FR import process

### PR DESCRIPTION
**Pull request by vpodzime**
We have switched to doing inventory refresh as part of the import
process because we needed to do a refresh by hostkey only for the
superhub. But this is now happening implicitly with the new
PostgreSQL notification handling mechanism. The CLI task is only
refreshing the lists of variables to show or not show in Mission
Portal. And that's the same on a superhub, a feeder or any other
hub. So we can now again just rely on the CLI task to be run as
part of an agent run.

### CODE REVIEW

File: cfe_internal/enterprise/federation/federation.cf [hunk 1]
```diff
@@ -374,7 +361,7 @@ bundle agent federation_manage_files
         template_method => "mustache",
         edit_template => "$(this.promise_dirname)/../../../templates/federated_reporting/config.sh.mustache",
         template_data => mergedata(@(login), @(feeder_username), @(feeder), @(cf_version),
-                                   @(cfengine_enterprise_federation:inventory_refresh_cmd.cmd)),
+                                   parsejson('"inventory_refresh_cmd": ""')),
```
Comment (nickanderson) [inline cfe_internal/enterprise/federation/federation.cf]: That works? Isn't it missing `{` and `}`?

Comment (nickanderson) [reply cfe_internal/enterprise/federation/federation.cf]: bundle agent main
    {
      reports:
        "BEFORE:$(with)" with => string_mustache( "{{%-top-}}", mergedata( @(inventory_refresh_cmd.cmd) ));
        "AFTER: $(with)" with => string_mustache( "{{%-top-}}", mergedata( parsejson('"inventory_refresh_cmd": ""') ));
    }
    bundle agent inventory_refresh_cmd
    {
      vars:
          "cmd" data => parsejson('{"inventory_refresh_cmd":
                                    "$(sys.workdir)/httpd/php/bin/php blah"}');
    }

OUTPUT:

       error: Non-container from parsing JSON expression '": ""'
       error: Non-container from parsing JSON expression '": ""'
    R: BEFORE:{
      "inventory_refresh_cmd": "/home/nickanderson/.cfagent/httpd/php/bin/php blah"
    }
       error: Non-container from parsing JSON expression '": ""'
       error: Non-container from parsing JSON expression '": ""'
    R: AFTER: $(with)
       error: Non-container from parsing JSON expression '": ""'
       error: Non-container from parsing JSON expression '": ""'
       error: Non-container from parsing JSON expression '": ""'
       error: Non-container from parsing JSON expression '": ""'

Comment (vpodzime) [reply cfe_internal/enterprise/federation/federation.cf]: Sigh... I wish `cf-promises cfe_internal/enterprise/federation/federation.cf` would tell me. :disappointed:

Comment (vpodzime) [reply cfe_internal/enterprise/federation/federation.cf]: Fixed and manually tested.

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4656/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4656/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4656/

**Comment by vpodzime** [comment]
> [Build Status](https://ci.cfengine.com//job/pr-pipeline/4656/)
> 
> **Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4656/

FR tests failed because they need to be adapted to this change.

**Comment by vpodzime** [comment]
> FR tests failed because they need to be adapted to this change.

https://github.com/cfengine/system-testing/pull/296

**Comment by vpodzime** [comment]
> [cfengine/system-testing#296](https://github.com/cfengine/system-testing/pull/296)

[Build Status](https://ci.cfengine.com/job/fr-tests/942/)

**Pull request merged**